packages/core/chrome/core-chrome-browser-internal/src/project_navigation/project_navigation_service.ts

```diff
@@ -243,13 +248,26 @@ export class ProjectNavigationService {
         next: ({ navigationTree, navigationTreeUI }) => {
           this.navigationTree$.next(navigationTree);
           this.navigationTreeUi$.next(navigationTreeUI);
-
           this.projectNavigationNavTreeFlattened = flattenNav(navigationTree);
 
-          // At initialization, we want to force the update of the active nodes, so 2 empty arrays []
-          // are not considered equal and we update the Observable value.
-          this.updateActiveProjectNavigationNodes({ forceUpdate: !initialised });
-          initialised = true;
+          // Verify if the current location is part of the navigation tree of
+          // the initiated solution. If not, we need to find the correct solution
+          const activeNodes = this.updateActiveProjectNavigationNodes();
+          let willChangeSolution = false;
+
+          if (activeNodes.length === 0) {
+            const solutionForCurrentLocation = this.findSolutionForCurrentLocation();
```

Comment:
Note: there is a good chance that all the logic regarding finding the correct solution for the current location will be removed. I still leave it for now until it is confirmed.
The goal is that the side nav is defined by the space you are in. I someone modifies the URL we should not try to change the side nav.